repository/src/test/java/uk/gov/hmcts/ccd/definition/store/database/RoutingDataSourceTest.java

```diff
@@ -0,0 +1,30 @@
+package uk.gov.hmcts.ccd.definition.store.database;
```

Comment:
I did add also an integration test for the /import controller to make sure all operations are sent to the master db. Unfortunately any integration test won't work. Because tests are single threaded. And this change relies on ThreadLocal. Which would require the test to use separate threads to work. So, in short, no idea how to test this with an integration test.

Reply:
is there an easy way to do that Mutlu without writing complex multhithread handling code in a test? If I spawn a new thread in a test, then the main thread has to wait for the spawned thread to complete before marking the test as completed. Things gets messy.
I hoped JUNIT offered support for this but couldn't find anything.